### ISSUE TITLE
Separate repository for BibleTime Mobile

### ISSUE THREAD

**Issue by jaakristioja** [bibletime/bibletime#157]
I'm thinking about removing BibleTime Mobile from the master branch and moving it to a separate repository, because keeping two frontends in the same repository just doesn't seem to be a good idea:

1. It complicates the build system, and will do so even more in the future.
2. Developers focusing on only one frontend face the risk of breaking the other frontend when changing the common files, especially because both targets can't be built and tested at the same time (one needs to reconfigure the build with `cmake -DBIBLETIME_FRONTEND=MOBILE` to switch between these two).
3. Git checkouts and source code packages will contain source code for both, and it might not be a good use of resources.
4. Development of these two frontends will continue to diverge, possibly leading to conflicts in the future.

The simplest approach would be to just create a new repository for BibleTime Mobile, so that the two repositories duplicate the shared code. The downside of this approach is that relevant fixes in one repository might propagate to the other repository, because this would require some manual work. The upside is that one can freely work with one frontend without fearing to break compatibility with the other frontend.

Another possibility is to try to refactor the shared code to a third repository. This would hopefully avoid (most) code duplication and fixes in shared code will apply to both frontends. On the other hand, this might still require the extra discipline to keep the shared code compatible with both frontends, which is rather tedious and might not work in the long term due to completely different requirements of these frontends/platforms.

Any thoughts or ideas about this and how to proceed?

**Comment by gholmlund** [comment]
Actually the code does not need any refactoring to be separated. We build a bibletime_common library which consists of the backend and util source. So we could have repositories like:
  bibletime_common 
  bibletime_desktop
  bibletime_mobile
The mobile version uses bibletime_common and does not use any bibletime_desktop
I would like to wait on doing this until the new continuous scrolling code is merged to the main repository.

One small issue - Transifex. I would like to keep the mobile translations together with the desktop translations on the transifex.com web site. The mobile version has received several language translations by just being together with the desktop version.

**Comment by gholmlund** [comment]
One other thought about transifex. Splitting the common and desktop into two repositories will probably require the transifex.com site to have two sets of files for the source code. One for the common and another for the desktop code. I am not sure how to do that separation. Would there be any loss of translations?

**Comment by jaakristioja** [comment]
> Actually the code does not need any refactoring to be separated.

Actually it already does need a bit of refactoring. Namely, `src/util/directory.cpp` in bibletime_common already contains some `#ifdef BT_MOBILE` logic, hence a different bibletime_common is built for the mobile frontend.

> I would like to wait on doing this until the new continuous scrolling code is merged to the main repository.

I agree. There's no hurry to take immediate action. I just wanted to raise this issue so that we could first analyse and discuss this situation, because it is not a trivial issue. From the viewpoint of a desktop frontend developer, my current (selfish) preference is to have two repositories: one for each frontend. But this might not be an optimal solution overall.

> One small issue - Transifex. I would like to keep the mobile translations together with the desktop translations on the transifex.com web site. The mobile version has received several language translations by just being together with the desktop version.

Indeed, keeping these translations together might certainly help. I don't know whether there would be any loss of translations if we split the repositories.

**Comment by jaakristioja** [comment]
However, something we could do to make the current situation better without splitting any repositories, is to get rid of the `BT_MOBILE` macro and amend the build system so that both frontends can be built at the same time, e.g. `make desktop` and `make mobile`. I'm not sure whether `make all` should build both by default, or just the desktop frontend. There might also be a need for targets like `make tests`, `make common_tests`, `make desktop_tests` and `make mobile_tests` etc.

If both frontends can be built without reconfiguring (i.e. rerunning CMake with the `-DBIBLETIME_FRONTEND=` option), this will probably simplify the build system and alleviate the risk of breaking the other frontend.

This would not solve all issues I mentioned in [my first comment](https://github.com/bibletime/bibletime/issues/157#issue-363243203), but would be of great help nonetheless.

PS: As you might have noticed, I deleted and then recreated the [swordxx](https://github.com/bibletime/bibletime/tree/swordxx) branch. This was to port BibleTime to using a newer revision of Sword++. Only after pushing it to GitHub I realized that I had only actually tested the desktop frontend. Because the build system only builds one frontend, I'm not even sure that the mobile frontend in the swordxx branch compiles. If both would have been built by default, it might have helped me to avoid this.
